### ISSUE TITLE
Genomic Information Not Retrieved for Any Junction

### ISSUE THREAD

**Issue by DarioS** [VCCRI/Ularcirc#11]
No matter which junction I click on in Gene_View, when I click on the Junction_View tab I get the pop-up error "Cannot retrieve genomic information for this gene". Below the load database button in Setup tab, I see Hsapiens.UCSC.hg38, so it seems the annotations have successfully been loaded.

**Comment by davhum** [comment]
Hi Dario,

Can you confirm this is the case for the TwoSzabo data set? If TwoSzabo data sets works can you let me know if you analysing STAR/CIRI/CircExplorer output? If you are NOT using STAR can you copy paste an example of the gene name as it might be a lookup issue.

Thanks,
D

**Comment by DarioS** [comment]
Ah, I figured it out now. It happens when Annotate With Parental Gene is not checked. Perhaps the pop-up error should state this as a possible cause. Currently, it only advises to check a database was loaded in Setup tab, which it was.

I notice a couple of other issues. There's always a red Shiny warning message at the bottom of output:

![image](https://user-images.githubusercontent.com/631218/75508069-ff871c80-5a36-11ea-80a8-19266109cd69.png)

Also, one gene - MUC7 - has a circular RNA of length infinity. I attach input files to reproduce. [test.zip](https://github.com/VCCRI/Ularcirc/files/4265174/test.zip)

**Comment by davhum** [comment]
Good suggestions.

I have started to look at test data set. The MUC7 example doesn't look like a BSJ, but rather a forward splice junction. I need to work out why this enters chimeric output. Will have a solution implemented in next day or so.

**Comment by davhum** [comment]
Any chance you could attach a couple of sequences from fastq file. This will be useful sanity check. 
For example the following read IDs would be useful:


A00121:71:HFFY2DSXX:2:1106:11776:28745
A00121:71:HFFY2DSXX:2:1124:22236:36495
A00121:71:HFFY2DSXX:2:1146:19090:8844

**Comment by DarioS** [comment]
The pair of FASTQ files for testing is in the archive [testReads.zip](https://github.com/VCCRI/Ularcirc/files/4290568/testReads.zip) Trimmed using `cutadapt`.

**Comment by davhum** [comment]
When I blat each of those reads it shows that they are indeed canonical junctions (see image below). Given that the CIGAR strings in chimeric junction output also suggested that they are not chimeric suggests some sort of leaky non-chimeric reporting by in chimeric output of STAR aligner. 


There are a couple of things I would now like to do:
1) Modify Ularcirc to filter out these candidates. 
2) Chase up why STAR aligner reports these reads as chimeric. On this note are you OK if I pass the testReads.zip you generated as part of a issue on STAR aligner.  



![MUC7_blat_fastq](https://user-images.githubusercontent.com/4456671/75948857-b2a0bb80-5ef8-11ea-8ba7-3f3908b0ba8d.png)

In above image have labelled each forward read as F1 F2 F3 and each paired end read ar R1 R2 R3.

**Comment by DarioS** [comment]
Yes, please create a new issue for STAR aligner using this test data. I am happy if you avoid creating extra filtering in Ularcirc and just get it fixed at the origin. I plan to re-map the data set soon anyway, because the next version of STAR will support writing multi-mapping chimeras within the BAM file, which is useful for [arriba](https://arriba.readthedocs.io/en/latest/) and rearrangements involving immunoglobulin genes, so I am happy to wait a while.

**Comment by davhum** [comment]
[Submitted issue](https://github.com/alexdobin/STAR/issues/852) to STAR.
